syncopy/io/load_ft.py

```diff
@@ -0,0 +1,419 @@
+    # with the default dimord ['time', 'channel']
+    h5FileOut = h5py.File(AData.filename, mode="w")
+    ADset = h5FileOut.create_dataset("data",
+                                     dtype=np.float32,
+                                     shape=[nTotalSamples, nChannels])
+
+    pbar = tqdm(trl_refs, desc=f"{struct_name} - loading {nTrials} trials")
+    SampleCounter = 0   # trial stacking
+
+    # one swipe per trial
+    for tr in pbar:
+        ADset[SampleCounter:nSamples, :] = h5File[tr]
+        SampleCounter += nSamples
+    pbar.close()
+
+    AData.data = ADset
+
+    # -- trialdefinition --
+
+    nTr_rng = np.arange(nTrials)
+    sampleinfo = np.vstack([nTr_rng, nTr_rng + 1]).T * nSamples
+
+    offsets = []
+    # we need to look into the time vectors for each trial
+    for time_r in time_refs:
+        offsets.append(h5File[time_r][0, 0])
+    offsets = np.array(offsets)
+
+    trl_def = np.hstack([sampleinfo, offsets[:, None]])
+    AData.trialdefinition = trl_def
+
+    # each channel label is an integer array with shape (X, 1),
+    # where `X` is the number of ascii encoded characters
+    channels = [''.join(map(chr, h5File[cr][:, 0])) for cr in chan_refs]
+    AData.channel = channels
+
+    # -- additional fields --
+
+    # this is the most experimental part
+    AData.info = {}
+    for field in add_fields:
+        if field not in h5Group:
+            msg = f"Could not find additional field {field} in {struct_name}"
+            SPYWarning(msg)
+            continue
+
+        # again an array of hdf5 object references
+        af_refs = h5Group[field][:, 0]
+
+        AData.info[field] = []
+        for af_ref in af_refs:
+            # here would be more parsing needed
+            # if we want to generally (strings, numbers, ..)
+            # support this
+            AData.info[field] = h5File[af_ref]
+
+    return AData
+
+
+def _read_dict_structure(structure, add_fields=None):
+
+    '''
+    Local helper to parse a single FT structure
+    and return an `~syncopy.AnalogData` object
+
+    Only for for Matlab data format version < 7.3
+    which was opened via scipy.io.loadmat!
+
+    This is the translation from FT to Syncopy:
+
+    FT     Syncopy
+
+    label - channel
+    trial - trial
+    time  - time
+
+    optional:
+    fsample - samplerate
+
+    cfg - X
+
+    Each trial in FT has nChannels x nSamples ordering,
+    Syncopy has nSamples x nChannels
+    '''
+    
+    # nTrials = structure["trial"].shape[0]
+    trials = []
+        
+    # 1st trial as reference
+    nChannels, nSamples = structure['trial'][0].shape
+
+    # check equal trial lengths
+    for trl in structure['trial']:
+
+        if trl.shape[-1] != nSamples:
+            lgl = 'Trials of equal lengths'
+            actual = 'Trials of unequal lengths'
+            raise SPYValueError(lgl, varname="load .mat", actual=actual)
+
+        # channel x sample ordering in FT
+        trials.append(trl.T.astype(np.float32))
```

Comment:
Same as above - can we be sure the mat file always contains 32bit floats?

Reply:
They actually have 64bit float, but I got no problems with the test dataset provided by Mohsen